pro_tes/middleware/middleware_pipeline.py

```diff
@@ -0,0 +1,173 @@
+"""Set up the middleware pipeline."""
+import importlib
+import logging
+from typing import Type
+import requests
+
+from pro_tes.exceptions import InvalidMiddleware
+from pro_tes.middleware.middleware import AbstractMiddleware
+
+logger = logging.getLogger(__name__)
+
+
+class MiddlewarePipeline:
+    """
+    Middleware Pipeline.
+
+    A class for managing and processing incoming requests through a sequence of
+    middleware components.
```

Comment:
We can see that it's a class. Start docstrings with a verb in imperative form. E.g.:

> Apply a sequence of middlewares to requests.